ext/ddtrace_profiling_native_extension/private_vm_api_access.c

```diff
@@ -23,13 +23,186 @@
 //
 // Note that beyond returning the rb_thread_struct*, rb_check_typeddata() raises an exception
 // if the argument passed in is not actually a `Thread` instance.
-static inline struct rb_thread_struct *thread_struct_from_object(VALUE thread) {
+static inline rb_thread_t *thread_struct_from_object(VALUE thread) {
   static const rb_data_type_t *thread_data_type = NULL;
   if (thread_data_type == NULL) thread_data_type = RTYPEDDATA_TYPE(rb_thread_current());
 
-  return (struct rb_thread_struct *) rb_check_typeddata(thread, thread_data_type);
+  return (rb_thread_t *) rb_check_typeddata(thread, thread_data_type);
 }
 
 rb_nativethread_id_t pthread_id_for(VALUE thread) {
   return thread_struct_from_object(thread)->thread_id;
 }
+
+// This is temporary just to break up implementation into two PRs and will be reverted in
+// https://github.com/DataDog/dd-trace-rb/pull/2000
+#ifdef TEMPORARY_SKIP_OLDER_RUBIES
+int ddtrace_rb_profile_frames(VALUE thread, int start, int limit, VALUE *buff, int *lines) { return 0; }
+
+#else
+
+// -----------------------------------------------------------------------------
+// The sources below are modified versions of code extracted from the Ruby project.
+// The Ruby project copyright and license follow:
```

Comment:
Were there modifications to the source code we copied here?
If so, could you leave some inline comments around the code lines that were modified?

Reply:
Hopefully addressed in 4b7ee20221ded8a82cfa37312bdabac7eacc9281:

* Added a note to document that every function has a list of changes (most of them are `Modifications: none`)
* Added an inline comment in `ddtrace_rb_profile_frames` to point at what changed from the upstream version (added `thread` argument)